core/src/trezor/crypto/__init__.py

```diff
@@ -9,6 +9,11 @@
     random,
 )
 
+try:
+    from trezorcrypto import elligator2  # noqa: F401
+except Exception:
+    pass
+
```

Comment:
this code would run on production firmware
maybe hide the try-except inside an `if __debug__:` at least? and add a comment that this is for an upcoming feature?

Reply:
For now, it will be only used in the new THP. So, it might give more sense to hide it this way:
```
if utils.THP_VERSION == 3:
    from trezorcrypto import elligator2
```
What do you think?

Reply:
It's cleaner but it is not doing the same thing .)

We need to statically exclude the branch, so that it does not end up compiled into production firmwares. `if __debug__` will do that, because the bytecode compiler can statically evaluate it to `if False` and completely omit the branch.

(we also have custom code that does search-and-replace of `if utils.BITCOIN_ONLY` and a couple other things, but `THP_VERSION` is not one of those things.)

I consider `if __debug__` to be a hack that we'll drop at some point -- but it doesn't seem worth more work into a clean solution.

Reply:
The goal wasn't to do the same thing. 

Either what you're suggesting doesn't make sense to me, or I don't understand the problem you are trying to solve. If you remove `if __debug__`, the branch will still be included in firmwares that don't use the new THP.

Reply:
the problem is, I want to statically exclude the import statement from compiled firmwares (for the time being)
`if THP_VERSION` is _dynamically_ excluding it, i.e., the if-condition (and the import statement and the reference to the module) becomes part of compiled bytecode.

(i mean, it's not as if this has any practical effect either way, but, well. it's cleaner)

`if __debug__` is a fancier version of `if False` that still lets us run the code when we need it -- with the expectation that once we enable THP, we enable it for all models at once.
in other words, the import will become unconditional

do you expect that we will want the import to be conditional at some point in the future? like, say, TS3 with the old protocol but T3W1 with THP?

if so, how about:
```suggestion
if __debug__:  # exclude from built firmwares until THP is ready
    if utils.THP_VERSION == 3:
        from trezorcrypto import elligator2
```
why version 3 btw?

(FWIW, the conditional import will cause problems with the typechecker, but we'll cross _that_ bridge when we come to it)

Reply:
See https://github.com/trezor/trezor-firmware/pull/3435/commits/e92595bff754a852b52cafb73bfedf8ab04c2e01.